src/main/java/Parser.java

```diff
@@ -0,0 +1,155 @@
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Responsible for parsing the user's input and processing the desired command.
+ */
+public class Parser {
+
+    /**
+     * Reads the user input and return it.
+     *
+     * @param input An input String.
+     * @param ui A Ui object.
+     * @param tasks A TaskList object.
+     * @param storage A Storage object.
+     * @throws DukeException When user input cannot be understood.
+     */
+    public static void parseInput(String input, Ui ui, TaskList tasks, Storage storage) throws DukeException, IOException {
+        String[] parsed = input.split(" ", 2);
+        String keyword = parsed[0];
+        String body = "";
+        if (parsed.length > 1) {
+            body = parsed[1];
+        }
+        switch (keyword) {
+        case "list":
+            tasks.printList(ui);
+            break;
+        case "done":
+            if (isValidSize(body, tasks)) {
+                tasks.markTaskDone(getNumber(body), ui);
+            } else {
+                throw new DukeException("Invalid number");
+            }
+            break;
+        case "delete":
+            if (isValidSize(body, tasks)) {
+                tasks.deleteTask(getNumber(body), ui);
+            } else {
+                throw new DukeException("Invalid number");
+            }
+            break;
+        case "todo":
+            if (isValidFormat(body, Task.Type.TODO)) {
+                tasks.addTask(new Todo(body), ui);
+            } else {
+                throw new DukeException("Invalid task format");
+            }
+            break;
+        case "deadline":
+            if (isValidFormat(body, Task.Type.DEADLINE) && isValidTime(body, Task.Type.DEADLINE)) {
+                tasks.addTask(new Deadline(getDescription(body), getTime(body, Task.Type.DEADLINE)), ui);
+            } else {
+                throw new DukeException("Invalid task format");
+            }
+            break;
+        case "event":
+            if (isValidFormat(body, Task.Type.EVENT) && isValidTime(body, Task.Type.EVENT)) {
+                tasks.addTask(new Event(getDescription(body), getTime(body, Task.Type.EVENT)), ui);
+            } else {
+                throw new DukeException("Invalid task format");
+            }
+            break;
+        case "find":
+            tasks.findTask(body, ui);
+        case "bye":
+            break;
+        default:
+            throw new DukeException("Invalid command");
+        }
+        storage.writeFile(tasks);
+    }
+
+    /**
+     * Reads the user input and return it.
+     *
+     * @param input An input String.
+     * @return A boolean whether input is bye.
+     */
+    public static boolean isBye(String input){
+        return input.split(" ", 2)[0].equals("bye");
```

Comment:
just a small suggestion, maybe you can mention that input.split(...)[0] is the firstWord of the input by creating a variable so it's slightly clearer!